main.go

```diff
@@ -108,30 +109,35 @@ func main() {
 					return
 				}
 
-				// validate the cluster ids contained in the remote write request
-				clusterId, err := remotewrite.ValidateRequest(remoteWriteRequest)
-				if err != nil {
-					log.Printf("error validating remote write request: %v", err)
-					w.WriteHeader(http.StatusForbidden)
-					return
-				}
-
-				log.Println(fmt.Sprintf("remote write request received from %v", clusterId))
+				clusterId := "<unknown>"
+				if !remotewrite.IsMetadataRequest(remoteWriteRequest) {
+					// validate the cluster ids contained in the remote write request
+					clusterId, err = remotewrite.ValidateRequest(remoteWriteRequest)
+					if err != nil {
+						log.Printf("error validating remote write request: %v", err)
+						w.WriteHeader(http.StatusForbidden)
+						return
+					}
 
-				if tokenVerifier.Enabled() {
-					token := tokenVerifier.GetAuthenticationToken(r)
-					if token != "" {
-						err = tokenVerifier.ValidateToken(parsedTokenVerificationUrl, clusterId, token)
-						if err != nil {
-							log.Println(fmt.Sprintf("error validating auth token from %v: %v", clusterId, err.Error()))
-							w.WriteHeader(http.StatusUnauthorized)
+					log.Println(fmt.Sprintf("remote write request received from %v", clusterId))
+
+					if tokenVerifier.Enabled() {
+						token := tokenVerifier.GetAuthenticationToken(r)
+						if token != "" {
```

Comment:
I thik this part can be reordered to avoid the if/else by writing it as:
if token == "" {
  log, set headers and return
}

err = token validate
if err != nil {
  log, set headers and return
}